### PR TITLE
Do not send a notification if a Pokemon has expired

### PR DESCRIPTION
**Pull request by JackDoan**
## Description

Sometimes a notification can persist/exist even if the Pokemon in question has despawned. This helps prevent that.

## Motivation and Context

Addresses issue #2941 

## How Has This Been Tested?

Notifications for Rattata near me are still sent.
I'm not entirely sure what causes phantom notifications, but this should help prevent them.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
     Caveat: 'sendNotification()' is now coupled with the Pokemon marker. It probably should be refactored into something like sendPokemonNotification, but I haven't seen a style guide for this project yet.  ¯\_(ツ)_/¯ 
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by PathToLife** [comment]
Does this solve the notification spamming issue?
Case: When pokemon expires, chrome spams the pokemon found notification over 30 times till page refreshed.
Thanks, =) I don't wanna delve into it if you've fixed it!

**Comment by JackDoan** [comment]
@PathToLife It should! I haven't proven it yet, and that spamming may actually indicate a deeper underlying issue, but this should _probably_ fix it.

**Comment by MRokas** [comment]
There's also spam whenever Lured Pokemon spawns.

**Comment by Scyntrus** [comment]
I feel like this check should happen earlier in the code. The way you have implemented right now means that the audio.play() might still occur, and that's much more annoying than the notification.

**Comment by JackDoan** [comment]
@Scyntrus Nice catch! I've moved the check to account for this.

**Comment by invisiblek** [comment]
Would it make more sense to short circuit setupPokemonMarker entirely if its expired? I see no reason in doing anything in this method if the thing is expired.

**Comment by JackDoan** [comment]
@invisiblek Good thinking! I've adjusted my change to account for this.

File: static/js/map.js [hunk 1]
```diff
@@ -602,9 +602,9 @@ function setupPokemonMarker(item, skipNotification, isBounceDisabled) {
   if (notifiedPokemon.indexOf(item.pokemon_id) > -1) {
     if (!skipNotification) {
       if (Store.get('playSound')) {
-        audio.play();
```
Comment (invisiblek) [inline static/js/map.js]: hmm, the logic here seems off now. Its only going to notify if playSound is enabled

Comment (JackDoan) [inline static/js/map.js]: oops! I've fixed this.

File: static/js/map.js [hunk 2]
```diff
@@ -602,7 +602,7 @@ function setupPokemonMarker(item, skipNotification, isBounceDisabled) {
...
+      audio.play();
```
Comment (andrewda) [inline static/js/map.js]: Why did you remove the indentation? Please re-add 2 spaces.
